src/groupeddataframe/groupeddataframe.jl

```diff
@@ -323,6 +323,20 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function _dict_to_tuple(key::Dict{T, S}, gd) where {T, S}
+    t = ntuple(length(gd.cols)) do i 
+        key[T(_names(gd)[i])]
+    end
+end
+
+function Base.to_index(gd::GroupedDataFrame, key::Dict{T, S}) where {T<:Union{Symbol, <:AbstractString}, S}
```

Comment:
We should not allow mixing `Symbol`s and strings I think:
```suggestion
function Base.to_index(gd::GroupedDataFrame, key::Union{Dict{Symbol},Dict{<:AbstractString}})
```